clinica/iotools/utils/data_handling.py

```diff
@@ -57,18 +57,27 @@ def create_merge_file(
         if not path.isdir(caps_dir):
             raise IOError("The path to the CAPS directory is wrong")
 
-    if not os.path.isfile(path.join(bids_dir, "participants.tsv")):
-        raise IOError("participants.tsv not found in the specified BIDS directory")
-    participants_df = pd.read_csv(path.join(bids_dir, "participants.tsv"), sep="\t")
-
     sessions, subjects = get_subject_session_list(
-        bids_dir, ss_file=tsv_file, use_session_tsv=True
+        bids_dir, ss_file=tsv_file, use_session_tsv=(not ignore_sessions_files)
     )
+
+    participants_df = pd.DataFrame(list(set(subjects)), columns=["participant_id"])
+
     sub_ses_df = pd.DataFrame(
         [[subject, session] for subject, session in zip(subjects, sessions)],
         columns=["participant_id", "session_id"],
     )
-    sub_ses_df.set_index(["participant_id", "session_id"], inplace=True)
+
+    try:
+        sub_ses_df.set_index(
+            ["participant_id", "session_id"], inplace=True, verify_integrity=True
+        )
+    except ValueError:
+        cprint(
+            "Found duplicate subject session pair. Please check subjects_sessions list for any misentry.",
+            lvl="error",
+        )
+        return
```

Comment:
I wonder why not simply run `pd.DataFrame(...).drop_duplicates().set_index(...)`, or do `pd.DataFrame(...)`, detect duplicates with `.duplicated()`, show a warning and then drop duplicates and set index, instead of a logging an error string followed by an early return?

From a UX perspective, it's probably less frustrating I believe.

Reply:
Well I suppose we could have it as only a warning. But I suppose if we got to this point then the dateset got corrupted somehow and it might not be the only issue. That was my reasoning

Reply:
It depends on the origin of the `sub_ses` dataframe I guess. If it comes from a path discovery mechanism, then it should be unique. However, human mistakes (like a duplicate line) are possible when coming from a custom file.

I am just thinking that, for the latter case, users might find it frustrating that Clinica fails and ask them to clean their mess first, instead of discarding duplicate pairs and eventually warning them if need be.

I am fine with either. Your call 👍

Reply:
I mean the "verify integrity" part was your suggestion (!) If you  think `.drop_duplicates().set_index(...)` is better, I can switch it to that.

Reply:
Let's keep it that way for now. If users were to complain, we know we can relax that a little with this alternative.